src/libstd/sys/unix/os.rs

```diff
@@ -111,7 +109,10 @@ pub fn getcwd() -> io::Result<PathBuf> {
                     return Err(error);
                 }
             }
-            n *= 2;
+
+            let cap = buf.capacity();
+            buf.set_len(cap);
+            buf.reserve(1);
```

Comment:
Could you add a comment here just saying we're letting the vector decide how to best resize itself?